OpenMP check fails using setup.py

Hi, I just cloned this repository and I'm following the installation instructions for Amber users. Specifically:
```bash
git clone https://github.com/amber-md/pytraj
cd pytraj
python setup.py install --prefix=$AMBERHOME
```
When I run `python setup.py install --prefix=$AMBERHOME` I get a TypeError:
```python
xxx@warthog:/pytraj (15 files, 41k) $ python setup.py install --prefix=$AMBERHOME
install_type 
1.0.8.dev0+86bb5af
build = False, install = False
libcpptraj_files None
Traceback (most recent call last):
  File "setup.py", line 126, in <module>
    tarfile=tarfile)
  File "/pytraj/scripts/base_setup.py", line 498, in get_ext_modules
    output_openmp_check = subprocess.check_output(['nm', libcpptraj_files[0]]).decode().split('\n')
TypeError: 'NoneType' object is not subscriptable
```
I think the variable `libcpptraj_files` is not being processed correctly because when I edit `scripts/base_setup.py` to print out `libcpptraj_files[0]`, I get the same TypeError, and `libcpptraj_files` is set to `None`.

My `$AMBERHOME` variable correctly points to my amber14gnu installation. Do I need Amber 16?

Possibly unrelated, but doing the manual installation of `libcpptraj.so` fails because `ld` can't find LAPACK/BLAS:

```bash
xxx@warthog:/pytraj/cpptraj (14 files, 111k) $ ./configure -shared -openmp gnu

Assuming 64 bit architecture.
Enabling position-independent code for generating shared library.
SANDER API will be used if found.
Using gnu compilers
Installing cpptraj binary to /home/dslochower/data/projects/QMFF/code/pytraj/cpptraj/bin
Using OpenMP

Testing C compiler:
  OK
Testing C++ compiler:
  OK
Testing Fortran compiler:
  OK
Checking BZLIB
  OK
Checking ZLIB
  OK
Checking NetCDF
  OK
Checking LAPACK/BLAS
./configure: line 134: ./testp: No such file or directory
  Error: Unable to compile using:
         g++  -Ixdrfile -o testp testp.cpp -llapack -lblas   -lgfortran  -w
  Please check your compiler settings or configure flags. Error follows:
  --------------------
/usr/bin/ld: cannot find -llapack
/usr/bin/ld: cannot find -lblas
collect2: error: ld returned 1 exit status
  --------------------
```
I have `lapack` and `blas` 3.4.2-5.el7 installed through my system package manager.

Hi @drroe, when trying to automatically install `libcpptraj` using `python setup.py install` in the `pytraj` root directory, I get the following output, suggesting to me that the `cpptraj` installation script is having difficulty finding `lapack` and `blas` even though I have them installed through my system package manager. Any thoughts?

```bash
xxx@warthog:~/data/projects/yyy/code/pytraj (16 files, 41k) $ python setup.py install
install_type 
1.0.8.dev0+86bb5af
build = False, install = True
can not find libcpptraj but found ./cpptraj folder, trying to reinstall it to ./cpptraj/lib/ 

command =  python scripts/install_libcpptraj.py  -openmp -amberlib
install libcpptraj from current ./cpptraj folder
build command:  bash configure -shared --with-netcdf=/home/xxx/anaconda3 --with-blas=/home/xxx/anaconda3 --with-bzlib=/home/xxx/anaconda3 --with-zlib=/home/xxx/anaconda3 -openblas -noarpack -amberlib -openmp  gnu || exit 1

Assuming 64 bit architecture.
Enabling position-independent code for generating shared library.
Using NetCDF in /home/xxx/anaconda3
Using BLAS in /home/xxx/anaconda3
Using BZIP2 in /home/xxx/anaconda3
Using ZLIB in /home/xxx/anaconda3
Using OpenBLAS
Not using ARPACK routines.
Using BLAS/LAPACK/ARPACK/NetCDF libraries in /home/xxx/amber14gnu
SANDER API will be used if found.
Using gnu compilers
Installing cpptraj binary to /run/media/xxx/data/projects/yyy/code/pytraj/cpptraj/bin
Using OpenMP

Testing C compiler:
  OK
Testing C++ compiler:
  OK
Testing Fortran compiler:
  OK
Checking BZLIB
  OK
Checking ZLIB
  OK
Checking NetCDF
configure: line 134: ./testp: No such file or directory
  Error: Unable to compile using:
         g++  -I/home/xxx/anaconda3/include -I/home/xxx/anaconda3/include -I/home/xxx/anaconda3/include -I/home/xxx/anaconda3/include -I/home/xxx/amber14gnu/include -Ixdrfile -o testp testp.cpp -L/home/xxx/anaconda3/lib -lnetcdf
  Please check your compiler settings or configure flags. Error follows:
  --------------------
/usr/bin/ld: warning: libmfhdf.so.0, needed by /home/xxx/anaconda3/lib/libnetcdf.so, not found (try using -rpath or -rpath-link)
/usr/bin/ld: warning: libdf.so.0, needed by /home/xxx/anaconda3/lib/libnetcdf.so, not found (try using -rpath or -rpath-link)
/usr/bin/ld: warning: libjpeg.so.8, needed by /home/xxx/anaconda3/lib/libnetcdf.so, not found (try using -rpath or -rpath-link)
/usr/bin/ld: warning: libhdf5_hl.so.10, needed by /home/xxx/anaconda3/lib/libnetcdf.so, not found (try using -rpath or -rpath-link)
/usr/bin/ld: warning: libhdf5.so.10, needed by /home/xxx/anaconda3/lib/libnetcdf.so, not found (try using -rpath or -rpath-link)
/usr/bin/ld: warning: libssl.so.1.0.0, needed by /home/xxx/anaconda3/lib/libnetcdf.so, not found (try using -rpath or -rpath-link)
/usr/bin/ld: warning: libcrypto.so.1.0.0, needed by /home/xxx/anaconda3/lib/libnetcdf.so, not found (try using -rpath or -rpath-link)
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Gopen1'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_STD_U16LE_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Dset_extent'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Pcreate'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Tget_native_type'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Aexists'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Pfill_value_defined'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Pset_layout'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Tis_variable_str'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Oopen'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Acreate1'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Tget_size'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5DSis_scale'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Fflush'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Fopen'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_NATIVE_INT_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5check_version'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Pset_chunk_cache'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5DSdetach_scale'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Pget_layout'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_STD_U16BE_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_STD_I8BE_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `SDstart'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Awrite'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Tget_member_type'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Aclose'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Gunlink'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Dget_create_plist'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_IEEE_F64LE_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Idec_ref'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Dget_space'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5DSget_scale_name'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Literate'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_STD_I16LE_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_STD_U64BE_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Pget_chunk'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Gcreate2'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `SDend'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Fget_create_plist'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Iget_name'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5P_CLS_DATASET_ACCESS_ID_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Tget_member_name'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Tget_member_value'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Sget_simple_extent_npoints'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Tset_cset'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Gget_objinfo'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `SDattrinfo'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Pset_link_creation_order'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_NATIVE_LLONG_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Tenum_insert'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5free_memory'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Tinsert'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Tget_member_offset'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Pset_obj_track_times'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_NATIVE_SHORT_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `SDfileinfo'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Dread'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Tset_strpad'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Fget_obj_ids'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Pget_chunk_cache'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Gget_objname_by_idx'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_NATIVE_UCHAR_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Pset_fclose_degree'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Tequal'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Pset_fletcher32'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Pset_deflate'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Aopen_by_name'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_STD_I32BE_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Aget_space'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Sget_simple_extent_ndims'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `SDreadattr'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_STD_U32BE_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_STD_U64LE_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_IEEE_F64BE_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Tset_size'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Fis_hdf5'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5P_CLS_GROUP_CREATE_ID_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5DSget_num_scales'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_STD_I8LE_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Dopen1'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Tarray_create1'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Fget_obj_count'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Adelete'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Aget_type'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Fcreate'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `SDselect'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Pclose'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_STD_I32LE_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Screate_simple'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Pset_fapl_core'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Tcommit1'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Pget_nfilters'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Sselect_hyperslab'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5DSset_scale'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5DSattach_scale'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_NATIVE_DOUBLE_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `SDreaddata'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Pset_fill_time'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Sget_simple_extent_dims'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Tget_array_dims1'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5LTopen_file_image'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5P_CLS_DATASET_XFER_ID_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_C_S1_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_STD_U8LE_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Lexists'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Pget_version'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Tvlen_create'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Dopen2'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Tcopy'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Dget_type'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_NATIVE_ULLONG_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Screate'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Gget_objtype_by_idx'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Pget_filter2'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_NATIVE_UINT_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Pset_cache'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_IEEE_F32LE_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_STD_I16BE_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Pget_link_creation_order'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Tget_super'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_NATIVE_FLOAT_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Pset_attr_creation_order'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Eset_auto2'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Dget_access_plist'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_STD_U8BE_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Pset_shuffle'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Aopen_name'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `SDgetchunkinfo'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Aread'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Tget_order'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `SDdiminfo'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5P_CLS_DATASET_CREATE_ID_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Tget_nmembers'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_STD_I64LE_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Gget_create_plist'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Aget_num_attrs'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Sclose'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Tget_array_ndims'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Pset_fill_value'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_STD_U32LE_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Sget_simple_extent_type'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5open'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5DSiterate_scales'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5get_libversion'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Pset_chunk'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Oclose'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5P_CLS_FILE_CREATE_ID_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Gmove'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Tclose'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Aopen_idx'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_NATIVE_SCHAR_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Dcreate1'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Iinc_ref'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Gopen2'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `SDgetfillvalue'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Pget_fill_value'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Tenum_create'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_NATIVE_USHORT_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `SDgetinfo'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Dwrite'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Dclose'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_IEEE_F32BE_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Tget_class'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Tcreate'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5P_CLS_FILE_ACCESS_ID_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Fclose'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Gget_num_objs'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Gclose'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Dcreate2'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5T_STD_I64BE_g'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `SDgetdimid'
/home/xxx/anaconda3/lib/libnetcdf.so: undefined reference to `H5Aget_name'
collect2: error: ld returned 1 exit status
  --------------------

make: *** No rule to make target `config.h', needed by `libcpptraj'.  Stop.
make sure to 'export CPPTRAJHOME=$CPPTRAJHOME'and 'export LD_LIBRARY_PATH=$CPPTRAJHOME/lib:\$LD_LIBRARY_PATH'then 'python ./setup.py install'
libcpptraj_files []
[]
Warning:  It seems that there is no libcpptraj. Please install it
xxx@warthog:~/data/projects/yyy/code/pytraj (16 files, 41k) $ 
```

--- a/scripts/base_setup.py
+++ b/scripts/base_setup.py
@@ -224,7 +224,7 @@ if not release:
 def do_what(pytraj_home):
     do_install = do_build = False
     # this checking should be here, after checking openmp and other stuff
-    if '--help' in sys.argv or '-h' in sys.argv:
+    if '--help' in sys.argv or '-h' in sys.argv or '--help-commands' in sys.argv:
         do_install = do_build = False
     else:
         if 'install' in sys.argv:

--- a/scripts/base_setup.py
+++ b/scripts/base_setup.py
@@ -222,25 +222,15 @@ if not release:
         a.close()
 
 def do_what(pytraj_home):
+    do_install = do_build = False
     # this checking should be here, after checking openmp and other stuff
-    if len(sys.argv) == 2 and sys.argv[1] == 'install':
-        do_install = True
-    elif len(sys.argv) == 3 and sys.argv[1] == 'install' and os.path.join('AmberTools',
-                                                                          'src') in pytraj_home:
-        # install pytraj in $AMBERHOME
-        # do not use pytraj_inside_amber here in we call `do_what()` before calling get_cpptraj_info()
-        # don't mess this up
-        # $(PYTHON) setup.py install $(PYTHON_INSTALL)
-        do_install = True
+    if '--help' in sys.argv or '-h' in sys.argv:
+        do_install = do_build = False
     else:
-        do_install = False
-
-    if len(sys.argv) == 2 and sys.argv[1] == 'build':
-        do_build = True
-    elif 'build_ext' in sys.argv and '--help' not in sys.argv:
-        do_build = True
-    else:
-        do_build = False
+        if 'install' in sys.argv:
+            do_install = True
+        if 'build' in sys.argv or 'build_ext' in sys.argv:
+            do_build = True
     return do_install, do_build
 
 def install_libcpptraj(openmp_flag, from_github=False, use_amberlib=True):
@@ -492,7 +482,6 @@ def get_ext_modules(cpptraj_info,
         if not libcpptraj_files:
             libcpptraj_files = try_updating_libcpptraj(cpptraj_info.home_env,
                     setup_task.do_install, setup_task.do_build, cpptraj_included, openmp_flag, use_amberlib)
-        print('libcpptraj_files', libcpptraj_files)
     
         try:
             output_openmp_check = subprocess.check_output(['nm', libcpptraj_files[0]]).decode().split('\n')

--- a/setup.py
+++ b/setup.py
@@ -51,7 +51,8 @@ cpptraj_home = os.environ.get('CPPTRAJHOME', '')
 use_pip = any('pip' in arg for arg in sys.argv)
 install_type = os.environ.get("INSTALLTYPE", "")
 check_flag(install_type)
-print('install_type', install_type)
+if install_type:
+    print('install_type', install_type)
 
 if not cpptraj_home and use_pip:
     # if pip, require to set CPPTRAJHOME
@@ -108,22 +109,25 @@ else:
 
 setenv_cc_cxx(cpptraj_info.ambertools_distro, extra_compile_args, extra_link_args)
 
-ext_modules = get_ext_modules(cpptraj_info=cpptraj_info,
-                pytraj_src=pytraj_src,
-                setup_task=setup_task,
-                is_released=is_released,
-                need_cython=need_cython,
-                cpptraj_included=cpptraj_included,
-                libcpptraj_files=libcpptraj_files,
-                openmp_flag=openmp_flag,
-                use_amberlib=use_amberlib,
-                cython_directives=cython_directives,
-                Extension=Extension,
-                extra_compile_args=extra_compile_args,
-                extra_link_args=extra_link_args,
-                define_macros=define_macros,
-                use_pip=use_pip,
-                tarfile=tarfile)
+if not do_build and not do_install:
+    ext_modules = []
+else:
+    ext_modules = get_ext_modules(cpptraj_info=cpptraj_info,
+                    pytraj_src=pytraj_src,
+                    setup_task=setup_task,
+                    is_released=is_released,
+                    need_cython=need_cython,
+                    cpptraj_included=cpptraj_included,
+                    libcpptraj_files=libcpptraj_files,
+                    openmp_flag=openmp_flag,
+                    use_amberlib=use_amberlib,
+                    cython_directives=cython_directives,
+                    Extension=Extension,
+                    extra_compile_args=extra_compile_args,
+                    extra_link_args=extra_link_args,
+                    define_macros=define_macros,
+                    use_pip=use_pip,
+                    tarfile=tarfile)
 
 setup_args = {}
 packages = [

